resolver/resolver.cc

```diff
@@ -4039,7 +4034,8 @@ class ResolveSignaturesWalk {
 
                 overloadSym.data(ctx)->setMethodVisibility(mdef.symbol.data(ctx)->methodVisibility());
                 overloadSym.data(ctx)->intrinsicOffset = mdef.symbol.data(ctx)->intrinsicOffset;
-                if (i != sigs.size() - 1) {
+                // We don't mark the last sig as overloaded to indicate that this is the last one in the chain.
+                if (i != sigs.size()) {
                     overloadSym.data(ctx)->flags.isOverloaded = true;
```

Comment:
More of an idle thought, but do you think it's worth having the overload bit on the method anymore for the methods with the `UniqueNameKind::Overload` names?

i.e., we could make it so that only the base `<U ...>` method symbol has `flags.isOverloaded`, and then _none_ of the overload methods have this bit set. As far as I can tell, the only place where we depend on that bit being set is in `guessOverload`, where we loop upwards to discover all the overload candidates.

Anyways, doesn't need to be changed now, just an idle thought for something that we might be able to simplify.

Reply:
I was using it to signal the tail of the linked list they form, but we could just as easily use lookup failure for that.